clang/include/clang/Basic/Diagnostic.h

```diff
@@ -946,6 +953,24 @@ class DiagnosticsEngine : public RefCountedBase<DiagnosticsEngine> {
     return (Level)Diags->getDiagnosticLevel(DiagID, Loc, *this);
   }
 
+  /// Diagnostic suppression mappings can be used to ignore diagnostics based on
+  /// the file they occur in. Mapping file is expected to be a special case list
+  /// with sections denoting diagnostic groups and `src` entries for globs to
+  /// suppress. `emit` category can be used to disable suppression. Longest glob
+  /// that matches a filepath takes precendence. For example:
+  ///   [unused]
+  ///   src:*clang/*
+  ///   src:*clang/foo/*=emit
+  ///   src:*clang/foo/bar/*
+  ///
+  /// Such a mappings file suppress all diagnostics produced by -Wunused in all
+  /// sources under `clang/` directory apart from `clang/foo/`. Diagnostics
```

Comment:
Why does the example start with * ?

Reply:
because filepaths are relative to execution root of clang, and not all projects/build systems run clang in the source root. so we can have prefixes like `../../` or `build/genfiles/` .

happy to drop that if you think that looks confusing.

Reply:
Would "*clang" catch "xclang" or does it assume "clang" must be a full path part?
Also, it would catch all clang directories, and not necessarily "the" clang/ directory.
It might be better to make this more explicit and not try to write it in a way that tries to catch everything.